cinema/views.py

```diff
@@ -24,28 +27,60 @@
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound("Method not allowed")
```

Comment:
The `raise NotFound("Method not allowed")` statement is misleading because it raises a `NotFound` exception instead of `MethodNotAllowed`. Consider using `MethodNotAllowed` for unsupported actions to provide a more accurate error response.